semgrep/tests/e2e/test_ci.py

```diff
@@ -135,29 +133,20 @@ def ci_mocks(base_commit, autofix):
         """
     ).lstrip()
 
-    with mock.patch.object(
-        GitlabMeta, "_fetch_branch_get_merge_base", return_value=base_commit
-    ):
-        # Mock rules
-        with mock.patch.object(
-            ConfigPath, "_make_config_request", mock.Mock(return_value=file_content)
-        ):
-            # Mock deployment_id
-            with mock.patch.object(
-                ScanHandler,
-                "_get_deployment_details",
-                mock.Mock(return_value=(DEPLOYMENT_ID, "org_name")),
-            ):
-                with mock.patch.object(auth, "is_valid_token", return_value=True):
-                    with mock.patch.object(
-                        ScanHandler, "autofix", mock.PropertyMock(return_value=autofix)
-                    ):
-                        with mock.patch.object(AppSession, "post"):
-                            yield
+    mocker.patch.object(ConfigPath, "_make_config_request", return_value=file_content)
+    mocker.patch.object(
+        ScanHandler, "_get_deployment_details", return_value=(DEPLOYMENT_ID, "org_name")
+    )
+    mocker.patch("semgrep.app.auth.is_valid_token", return_value=True)
+    mocker.patch.object(AppSession, "post")
+
+
+@pytest.fixture(params=[True, False], ids=["autofix", "noautofix"])
+def autofix(request, mocker):
+    mocker.patch.object(ScanHandler, "autofix", request.param)
```

Comment:
These same parameters were duplicated across multiple test functions. So here I extracted them to a fixture, and just by requesting the `autofix` fixture pytest will automatically parametrize a test function with autofix/noautofix for us.

This happened to change the test IDs though (local-autofix to autofix-local) which is why all those snapshot files are renamed.